src/WebJobs.Script/Rpc/ILanguageWorkerChannelManager.cs

```diff
@@ -0,0 +1,29 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Script.Diagnostics;
+
+namespace Microsoft.Azure.WebJobs.Script.Rpc
+{
+    public interface ILanguageWorkerChannelManager
+    {
+        Task InitializeWebhostChannelsAsync(IEnumerable<string> languages);
+
+        Task InitializeWebhostChannelAsync(string language);
+
+        ILanguageWorkerChannel GetWebHostChannel(string language);
+
+        Task ReloadEnvironmentVariablesAsync(string language);
```

Comment:
Is this manager maintaining a set of created/initialized channels? If so, then shouldn't this just be ReloadEnvironment, and internally it reloads environment for all channels? It's confusing because there are methods on this manager that create/return channels implying the caller now owns them, but other methods that seem to operate on the created channels.

Reply:
ReloadEnvironment --> is specifically called during specialization. Yes this manager is used for
 - Creating and maintaining worker channels at webhost level
 - Creating at jobhost level as well